binary_search_tree/tree.py

```diff
@@ -17,32 +17,121 @@ def __init__(self):
+                    add_helper(current.left)
+                else:
+                    current.left = TreeNode(key, value)
+            else:
+                if current.right is not None:
+                    add_helper(current.right)
+                else:
+                    current.right = TreeNode(key,value)
+        
+        add_helper(self.root)
 
     # Time Complexity: 
     # Space Complexity: 
     def find(self, key):
-        pass
+        if self.root == None:
+            return None
+
+        def find_helper(current):
+            if current.key == key:
+                return current.value
+            elif key < current.key:
+                if current.left is not None:
+                    return find_helper(current.left)
+            else:
+                if current.right is not None:
+                    return find_helper(current.right)
+        return find_helper(self.root)
 
     # Time Complexity: 
     # Space Complexity: 
     def inorder(self):
-        pass
+        return_list = []
+        current = self.root
+        if self.root == None:
+            return return_list
+
+        def in_order_helper(current):
+            if current.left is not None:
+                in_order_helper(current.left)
+            return_list.append({'key': current.key, 'value': current.value})
+            if current.right is not None:
+                in_order_helper(current.right)
+        
+        in_order_helper(current)
+        return return_list
 
     # Time Complexity: 
     # Space Complexity:     
     def preorder(self):
-        pass
+        return_list = []
+        current = self.root
+        if self.root == None:
+            return return_list
+
+        def pre_order_helper(current):
+            return_list.append({'key': current.key, 'value': current.value})
+            if current.left is not None:
+                pre_order_helper(current.left)
+            if current.right is not None:
+                pre_order_helper(current.right)
+        
+        pre_order_helper(current)
+        return return_list
 
     # Time Complexity: 
     # Space Complexity:     
     def postorder(self):
-        pass
+        return_list = []
+        current = self.root
+        if self.root == None:
+            return return_list
+
+        def post_order_helper(current):
+            if current.left is not None:
+                post_order_helper(current.left)
+            if current.right is not None:
+                post_order_helper(current.right)
+            return_list.append({'key': current.key, 'value': current.value})
+        
+        post_order_helper(current)
+        return return_list
 
     # Time Complexity: 
     # Space Complexity:     
     def height(self):
-        pass
+        #check for greatest number
+        #+1 going down stack, -1 going up stack
+        heights = 1
+        max_height = 0
+        current = self.root
+        if self.root == None:
```

Comment:
Same thing here with moving this up and bailing out immediately before you initialize the variables. Once again, mostly a preference and style thing, so not too big a deal in this case.